### PR TITLE
E2E for Headless services with globalnet

### PR DESCRIPTION
**Pull request by vthapar**
* Changes to support e2e for Globalnet+HeadlessServices
* Fixes to some existing bugs in e2e code
* Comment out some broker tests revealed by fixes in e2e code.

Depends On https://github.com/submariner-io/submariner-charts/pull/157

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

### CODE REVIEW

File: test/e2e/discovery/headless_services.go [hunk 1]
```diff
@@ -342,7 +329,7 @@ func verifyHeadlessSRVRecordsWithDig(f *framework.Framework, cluster framework.C
 					}
 
 					if !doesContain && shouldContain {
-						return false, fmt.Sprintf("expected execution result %q to contain %q", result, hostName), nil
+						return false, fmt.Sprintf("expected execution result %q to contain %q and %q", result, hostDNS, strconv.Itoa(int(port.Port))), nil
```
Comment (skitt) [inline test/e2e/discovery/headless_services.go]:
```suggestion
						return false, fmt.Sprintf("expected execution result %q to contain %q and %d", result, hostDNS, int(port.Port), nil
```

**Review by skitt — CHANGES_REQUESTED** [review]

File: test/e2e/discovery/headless_services.go [hunk 2]
```diff
@@ -39,41 +39,25 @@ var _ = Describe("[discovery] Test Headless Service Discovery Across Clusters",
 
 	When("a pod tries to resolve a headless service in a remote cluster", func() {
 		It("should resolve the backing pod IPs from the remote cluster", func() {
-			if !framework.TestContext.GlobalnetEnabled {
-				RunHeadlessDiscoveryTest(f)
-			} else {
-				framework.Skipf("Globalnet is enabled, skipping the test...")
-			}
+			RunHeadlessDiscoveryTest(f)
```
Comment (sridhargaddam) [inline test/e2e/discovery/headless_services.go:42]: This PR depends on the Submariner datapath PR which is currently under review - https://github.com/submariner-io/submariner/pull/1456

So we should not merge this until the Submariner PR is merged and references to Submariner are updated in go.mod
@vthapar can you please include "Depends on: https://github.com/submariner-io/submariner/pull/1456" in the commit message.

**Review by sridhargaddam — CHANGES_REQUESTED** [review]

File: test/e2e/discovery/headless_services.go (hunk 2 above)
Comment (vthapar) [reply test/e2e/discovery/headless_services.go:42]: @sridhargaddam This doesn't depend on datapath coz in LH we don't use curl for testing. We use `dig` which only checks if service is resolved, not connectivity.

**Review by tpantelis — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Comment by sridhargaddam** [comment]
@vthapar the Helm job is now failing on some ServiceDiscovery use-case (the previous RBAC issues are now resolved). Can you please take a look.
```
STEP: Creating serviceExport nginx-demo.e2e-tests-discovery-698nz on "cluster2"
STEP: Retrieving ServiceExport nginx-demo.e2e-tests-discovery-698nz on "cluster2"
panic: test timed out after 15m0s

goroutine 818 [running]:
testing.(*M).startAlarm.func1()
```

**Comment by vthapar** [comment]
@sridhargaddam Looks like charts is also missing RBAC for lighthouse-agent
``` 
I0707 12:23:07.325271       1 agent.go:393] updateExportedServiceStatus for (e2e-tests-discovery-698nz/nginx-demo) - Type: "Valid", Status: "False", Reason: "GlobalIngressIPRetrievalFailed", Message: "globalingressips.submariner.io \"nginx-demo\" is forbidden: User \"system:serviceaccount:submariner-operator:submariner-lighthouse-agent\" cannot get resource \"globalingressips\" in API group \"submariner.io\" in the namespace \"e2e-tests-discovery-698nz\""
```

**Comment by sridhargaddam** [comment]
Helm jobs are now passing!!
@skitt can you please take a relook at this PR. Thanks.

**Review by aswinsuryan — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**